integration-tests/utils/integration-helpers.js

```diff
@@ -32,24 +32,41 @@ const getTestWorkspaceName = () => `${testWorkspaceNamePrefix}${uuid.v4()}`
  * user access to a workspace. This function polls to check if the logged in user's pet service
  * account has read access to the given workspace's GCS bucket.
  */
-const waitForReadAccessToWorkspaceBucket = async ({ page, billingProject, workspaceName, timeout = defaultTimeout }) => {
+const waitForAccessToWorkspaceBucket = async ({ page, billingProject, workspaceName, timeout = defaultTimeout }) => {
   await page.evaluate(async ({ billingProject, workspaceName, timeout }) => {
+    const { workspace: { googleProject, bucketName } } = await window.Ajax().Workspaces.workspace(billingProject, workspaceName).details(['workspace'])
+
     const startTime = Date.now()
-    while (true) {
-      try {
-        await window.Ajax().Workspaces.workspace(billingProject, workspaceName).checkBucketReadAccess()
-        return
-      } catch (response) {
-        if (response.status === 403) {
-          if (Date.now() - startTime < timeout) {
-            // Wait 15s before retrying
-            await new Promise(resolve => setTimeout(resolve, 15 * 1000))
-            continue
+
+    const checks = [
+      // List objects
+      () => window.Ajax().Buckets.list(googleProject, bucketName, ''),
+      // Create object
+      () => {
+        const file = new File([''], 'permissions-check', { type: 'text/text' })
+        return window.Ajax().Buckets.upload(googleProject, bucketName, '', file)
+      },
+      // Delete object
+      () => window.Ajax().Buckets.delete(googleProject, bucketName, 'permissions-check'),
+    ]
+
+    for (const check of checks) {
```

Comment:
Hmm... instead of a for, would it be possible to use syntax along the lines of this, where we convert a list of check via map to something compatible with promise.all?

https://github.com/DataBiosphere/terra-ui/blob/dev/src/pages/workspaces/workspace/analysis/Analyses.js#L309

Reply:
The `for` syntax makes these checks serial, whereas `Promise.all(_.map(...))` would run them in parallel. In particular, we can't check if we can delete the test file before we've created it.

Reply:
Yeah, doesn't seem like there's a better way to do this in serial. I think the break/continue syntax will become cumbersome if this expands anymore though, but for now I'm fine with it.

Reply:
I tried rewriting this with `reduce`, but it didn't seem as clear and still included a `while(true)` and `break`.